### PR TITLE
Improve msg for missing libicu

### LINKED ISSUE #41350: When libicu is missing, message steers towards invariant mode not installing libicu
Install dotnet on Linux (eg., on Debian in my case just now)

```
dan@danmose3:~/git/performance$ tools/dotnet/x64/dotnet --info
Process terminated. Couldn't find a valid ICU package installed on the system. Set the configuration flag System.Globalization.Invariant to true if you want to run with no globalization support.
   at System.Environment.FailFast(System.String)
[2020/08/25 12:48:23][INFO]    at System.Globalization.GlobalizationMode.GetGlobalizationInvariantMode()
[2020/08/25 12:48:23][INFO]    at System.Globalization.GlobalizationMode..cctor()
[2020/08/25 12:48:23][INFO]    at System.Globalization.CultureData.CreateCultureWithInvariantData()
[2020/08/25 12:48:23][INFO]    at System.Globalization.CultureData.get_Invariant()
[2020/08/25 12:48:23][INFO]    at System.Globalization.TextInfo..cctor()
[2020/08/25 12:48:23][INFO]    at System.String.ToLowerInvariant()
[2020/08/25 12:48:23][INFO]    at System.Text.EncodingHelper.GetCharset()
[2020/08/25 12:48:23][INFO]    at System.Text.EncodingHelper.GetEncodingFromCharset()
[2020/08/25 12:48:23][INFO]    at System.ConsolePal.GetConsoleEncoding()
[2020/08/25 12:48:23][INFO]    at System.Console.get_OutputEncoding()
[2020/08/25 12:48:23][INFO]    at System.Console.CreateOutputWriter(System.IO.Stream)
[2020/08/25 12:48:23][INFO]    at System.Console.<get_Out>g__EnsureInitialized|25_0()
[2020/08/25 12:48:23][INFO]    at System.Console.get_Out()
[2020/08/25 12:48:23][INFO]    at Microsoft.DotNet.Cli.Utils.Reporter.Reset()
[2020/08/25 12:48:23][INFO]    at Microsoft.DotNet.Cli.Utils.Reporter..cctor()
[2020/08/25 12:48:23][INFO]    at Microsoft.DotNet.Cli.Utils.Reporter.get_Output()
[2020/08/25 12:48:23][INFO]    at Microsoft.DotNet.Cli.Program.PrintInfo()
[2020/08/25 12:48:23][INFO]    at Microsoft.DotNet.Cli.Program.ProcessArgs(System.String[], Microsoft.DotNet.Cli.Telemetry.ITelemetry)
[2020/08/25 12:48:23][INFO]    at Microsoft.DotNet.Cli.Program.Main(System.String[])
```

Can we be more helpful here? We seem to be steering folks towards invariant mode, when the answer most users will want is likely to simply install libicu.

Yes, the installer likely earlier warned `Warning: Unable to locate libicu. Probable prerequisite missing; install libicu.` but that can be overlooked way back in the console output.

Could the message say something like eg?
```
Process terminated. Couldn't find a valid ICU package installed on the system. Please install libicu using your package manager and try again. Alternatively you can set the configuration flag System.Globalization.Invariant to true to run with no globalization support.
```
It's a little tricky because we don't know exactly what the package name is. Eg., on Debian `apt-get install libicu` did not work but `apt-get install libicu-dev` did work as did `apt-get install libicu63` in my case.

I compared Windows, where the message just says `Failed to load app-local ICU: <some.dll>` but in that case I assume the user has already opted into app-local, so they have some idea of what ICU is and where it ought to be. Whereas on Unix, many folks encountering this message will not know what ICU is and nor should they need to know.
cc @safern 

BTW we already improved this for libgdiplus: `System.DllNotFoundException: Unable to load shared library 'libgdiplus' or one of its dependencies. In order to help diagnose loading problems, consider setting the LD_DEBUG environment variable: liblibgdiplus: cannot open shared object file: No such file or directory`

### PR DESCRIPTION
**Pull request by danmoseley**
Fix #41350

### CODE REVIEW

**Comment by GrabYourPitchforks** [comment]
What, no fwlink? :)

**Comment by danmosemsft** [comment]
How do we normally handle fwlink ownership? It seemed like it would be easy to Bing.

File: src/libraries/System.Private.CoreLib/src/System/Globalization/GlobalizationMode.Unix.cs [hunk 1]
```diff
@@ -26,7 +26,8 @@ private static bool GetGlobalizationInvariantMode()
                     if (loaded == 0 && !OperatingSystem.IsBrowser())
                     {
                         string message = "Couldn't find a valid ICU package installed on the system. " +
-                                        "Set the configuration flag System.Globalization.Invariant to true if you want to run with no globalization support.";
+                                         "Please install libicu using your package manager and try again. " +
+                                         "Alternatively you can set the configuration flag System.Globalization.Invariant to true if you want to run with no globalization support.";
```
Comment (stephentoub) [inline src/libraries/System.Private.CoreLib/src/System/Globalization/GlobalizationMode.Unix.cs:30]: Pre-existing your change, but this doesn't need to be in the resx?

Comment (tarekgh) [reply src/libraries/System.Private.CoreLib/src/System/Globalization/GlobalizationMode.Unix.cs:30]: No we cannot have this in resx. Resources need globalization. If ICU is missing then all globalization would be broken. We usually use hardcoded message for the things we fail fast with. one example when there is something wrong in the builds which cause recursive lookup for resources. at that time we just fail fast with hardcoded message.

**Review by tarekgh — APPROVED** [review]

File: src/libraries/System.Private.CoreLib/src/System/Globalization/GlobalizationMode.Unix.cs (hunk 1 above)
Comment (stephentoub) [reply src/libraries/System.Private.CoreLib/src/System/Globalization/GlobalizationMode.Unix.cs:30]: Can we add a comment here to that effect, highlighting that it's an explicit choice to not put this in resources?

**Comment by GrabYourPitchforks** [comment]
@danmosemsft Ownership belongs to the division as a whole. It's primarily useful for scenarios where: (a) we can't easily describe in a single sentence what corrective measures to take; or (b) we can't localize the message itself but we can point the fwlink to a localized landing page.

**Comment by danmosemsft** [comment]
Apparently aka.ms is the new fwlink. Added one pointing to https://docs.microsoft.com/en-us/dotnet/core/run-time-config/globalization.

**Comment by danmosemsft** [comment]
errors were https://github.com/dotnet/runtime/issues/47562 and https://github.com/dotnet/runtime/issues/47561

**Comment by danmosemsft** [comment]
Test failures are https://github.com/dotnet/runtime/issues/47561

**Review by safern — COMMENTED** [review]
LGTM

**Pull request merged**